### PR TITLE
vectorized triangulation angle computation

### PR DESCRIPTION
**Pull request by johnwlambert**
Used in ViewGraphEstimators.

### CODE REVIEW

File: gtsfm/densify/mvs_utils.py [hunk 1]
```diff
@@ -44,6 +45,49 @@ def calculate_triangulation_angle_in_degrees(
     return geometry_utils.compute_relative_unit_translation_angle(Unit3(ray_1), Unit3(ray_2))
 
 
+def calculate_triangulation_angles_in_degrees(
```
Comment (ayushbaid) [inline gtsfm/densify/mvs_utils.py:48]: Do we need the non-vectorized function anywhere?

File: gtsfm/densify/mvs_utils.py [hunk 2]
```diff
@@ -44,6 +45,49 @@ def calculate_triangulation_angle_in_degrees(
...
+    N = points_3d.shape[0]
+    # ensure broadcasting is in the correct direction
+    rays1 = points_3d - camera_center_1.reshape(1, 3)
+    rays2 = points_3d - camera_center_2.reshape(1, 3)
+
+    # normalize rays to unit length
+    rays1 /= np.linalg.norm(rays1, axis=1).reshape(N, 1)
+    rays2 /= np.linalg.norm(rays2, axis=1).reshape(N, 1)
+
+    dot_products = np.multiply(rays1, rays2).sum(axis=1)
```
Comment (ayushbaid) [inline gtsfm/densify/mvs_utils.py:84]: Why not use `np.dot`?

**Review by ayushbaid — APPROVED** [review]
LGTM except minor comments

File: gtsfm/densify/mvs_utils.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/densify/mvs_utils.py:48]: Thanks for taking a look @ayushbaid. I'm using it in another branch, and wanted to keep the other branch lean. COLMAP relies pretty heavily on minimum triangulation angles -- we can discuss more tomorrow.

File: gtsfm/densify/mvs_utils.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/densify/mvs_utils.py:84]: rays1 and rays2 are (N,3) and (N,3), so if we do `rays1 @ rays2.T` we get a `N x N` matrix, and only the diagonal is useful. This way we keep things efficient.

**Pull request merged**